src/components/Carousel.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="carousel">
+    <CarouselSlide  v-for="(slide, index) in slides" :key="slide" :index="index" :visibleSlide="visibleSlide">
+      <img :src="getImagePath(slide)">
+    </CarouselSlide>
+  </div>
+</template>
+
+<script>
+import CarouselSlide from '@/components/CarouselSlide.vue';
+export default {
+  name : 'CarouselComp',
+  data(){
+    return{
+      slides:[
+        '1.png',
+        '2.png',
+        '3.png',
+        '4.png',
+        '5.png',
+        '6.png'
+      ],
+      visibleSlide : 0
+    }
+  },
+  computed:{
+    getSlideLength(){
+      return this.slides.length
+    }
+  },
+  methods:{
+    getImagePath(image) {
+      return require(`@/assets/carousel/${image}`);
+    },
+    slideTransition(){
+      if(this.visibleSlide >= this.getSlideLength - 1){
+        this.visibleSlide = 0;
+      }
+      else{
+        this.visibleSlide++;
+      }
+    }
+  },
+  components:{
+    CarouselSlide
+  },
+  mounted(){
+    setInterval(this.slideTransition, 6000);
+  }
```

Comment:
the proper way to order these functions are: name, components, data, mounted,created, methods.